DDP-4564: Manage cookie preferences

marat-mamytov opened the pull request:
Current PR adds ability to manage cookies preferences.

**AC1, AC2**
When participant clicks on “Manage Preferences” in the cookie banner 
![image](https://user-images.githubusercontent.com/54633385/90137692-ce599900-dd97-11ea-9479-5c7baa44bfe2.png)
the modal “Manage Cookie Preferences” is displayed
![image](https://user-images.githubusercontent.com/54633385/90137809-f47f3900-dd97-11ea-90d8-96ce675be622.png)

**AC3**
There are 3 tabs:  “Your Privacy”, “Functional Cookies”, “Analytical Cookies”
For example:
![image](https://user-images.githubusercontent.com/54633385/90138262-91da6d00-dd98-11ea-9a91-0d3ff68fe89e.png)

But any additional categories of cookies can be added to the `type CookiesTypes` and listed in the project configuration.

**AC4**
Clicking on the “Submit Preferences” button -> submits the accepted or rejected cookies -> user preferences set in the Local Storage under `prion_cookies_consent` cookie + dismisses the modal and the underlying cookie banner
![image](https://user-images.githubusercontent.com/54633385/90138537-f8f82180-dd98-11ea-8f07-72a6201a6c87.png)

**AC5**
Click on the “X” -> dismiss the modal, not changes apply

**AC6**
Click on the “Privacy Policy” link -> launch the Privacy Policy page in the modal
![image](https://user-images.githubusercontent.com/54633385/90138834-5e4c1280-dd99-11ea-9647-8b3f363ce285.png)

**AC7**
The cookie preferences selected should persist across sessions since set in the Local Storage

**AC8**
On mobile view the modal takes up the whole view port
![image](https://user-images.githubusercontent.com/54633385/90139063-ac611600-dd99-11ea-870f-b8facc502599.png)

**AC9**
Added  a button to launch this cookie preference panel in the Privacy Policy page
![image](https://user-images.githubusercontent.com/54633385/90139218-e205ff00-dd99-11ea-85b8-11fa0f0f4e87.png)

**AC10**
Added logger to the cookies consent update event

charlottedye committed:
Initial work setting up prion frontend
Initial version of main prion pages, consent survey, and beginning of medical questionnaire: DDP-4083DDP-4110, DDP-4118, DDP-4120, DDP-4091, DDP-4096, DDP-4097, DDP-4098, DDP-4099, DDP-4092, and some of DDP-4282

charlottedye committed:
Initial work setting up prion frontend
Initial version of main prion pages, consent survey, and beginning of medical questionnaire: DDP-4083DDP-4110, DDP-4118, DDP-4120, DDP-4091, DDP-4096, DDP-4097, DDP-4098, DDP-4099, DDP-4092, and some of DDP-4282

charlottedye committed:
Initial work setting up prion frontend
Initial version of main prion pages, consent survey, and beginning of medical questionnaire: DDP-4083DDP-4110, DDP-4118, DDP-4120, DDP-4091, DDP-4096, DDP-4097, DDP-4098, DDP-4099, DDP-4092, and some of DDP-4282

charlottedye committed:
Initial work setting up prion frontend
Initial version of main prion pages, consent survey, and beginning of medical questionnaire: DDP-4083DDP-4110, DDP-4118, DDP-4120, DDP-4091, DDP-4096, DDP-4097, DDP-4098, DDP-4099, DDP-4092, and some of DDP-4282

zyme committed:
Adding prototype of running frontend in GAE

zyme committed:
Adding prototype of running frontend in GAE

zyme committed:
Switching to python27, which makes no sense but works

charlottedye committed:
Got Prion on Pepper to run successfully
DDP-4207

charlottedye committed:
Got Prion on Pepper to run successfully
DDP-4207

charlottedye committed:
Got Prion on Pepper to run successfully
DDP-4207

charlottedye committed:
Got Prion on Pepper to run successfully
DDP-4207

charlottedye committed:
DDP-4284: Got first page of medical questionnaire working except for substituting name

charlottedye committed:
DDP-4284: Got first page of medical questionnaire working except for substituting name

charlottedye committed:
Miscellaneous styling changes to symptomatic page 2 of Prion medical questionnaire
 DDP-4284

charlottedye committed:
Miscellaneous styling changes to symptomatic page 2 of Prion medical questionnaire
 DDP-4284

charlottedye committed:
DDP-4284: Initial version of first few pages of Prion medical questionnaire

charlottedye committed:
DDP-4284: Initial version of first few pages of Prion medical questionnaire

charlottedye committed:
DDP-4284: Prion medical questionnaire: implemented symptomatic page 4 and final page for all versions, made various text corrections, worked on styling, and miscellaneous refactoring.

charlottedye committed:
DDP-4284: Prion medical questionnaire: implemented symptomatic page 4 and final page for all versions, made various text corrections, worked on styling, and miscellaneous refactoring.

charlottedye committed:
DDP-4284: Prion medical questionnaire: font and text changes to better match existing version of Prion, fixed some spacing issues, made checkboxes look more like existing version in Prion, and miscellaneous cleanup

charlottedye committed:
DDP-4284: Prion medical questionnaire: font and text changes to better match existing version of Prion, fixed some spacing issues, made checkboxes look more like existing version in Prion, and miscellaneous cleanup

charlottedye committed:
DDP-4336: Cleaned up unneeded email verification code for Prion since new Prion won't need to have email verification.

charlottedye committed:
DDP-4336: Cleaned up unneeded email verification code for Prion since new Prion won't need to have email verification.

charlottedye committed:
DDP-4336: Cleaned up unneeded email verification code for Prion since new Prion won't need to have email verification.

charlottedye committed:
DDP-4336: Cleaned up unneeded email verification code for Prion since new Prion won't need to have email verification.

charlottedye committed:
DDP-4336: Finished getting rid of leftovers from email verification: got rid of text for email verification prompt page, changed header and welcome components to configure Auth0 to redirect to start-study page after account creation, and miscellaneous cleanup

charlottedye committed:
DDP-4336: Finished getting rid of leftovers from email verification: got rid of text for email verification prompt page, changed header and welcome components to configure Auth0 to redirect to start-study page after account creation, and miscellaneous cleanup

charlottedye committed:
DDP-4336: Finished getting rid of leftovers from email verification: got rid of text for email verification prompt page, changed header and welcome components to configure Auth0 to redirect to start-study page after account creation, and miscellaneous cleanup

charlottedye committed:
DDP-4336: Finished getting rid of leftovers from email verification: got rid of text for email verification prompt page, changed header and welcome components to configure Auth0 to redirect to start-study page after account creation, and miscellaneous cleanup

charlottedye committed:
DDP-4132: Created study listing page and miscellaneous cleanup

charlottedye committed:
DDP-4132: Created study listing page and miscellaneous cleanup

charlottedye committed:
DDP-4132: Created study listing page and miscellaneous cleanup

charlottedye committed:
DDP-4132: Created study listing page and miscellaneous cleanup

charlottedye committed:
DDP-4132: Made study listing page be sorted alphabetically by study name by default

charlottedye committed:
DDP-4132: Made study listing page be sorted alphabetically by study name by default

charlottedye committed:
DDP-4127, set up dashboard: changed dashboard title and descriptive text and started working on adding study listing page to dashboard just below activities

charlottedye committed:
DDP-4127, set up dashboard: changed dashboard title and descriptive text and started working on adding study listing page to dashboard just below activities

charlottedye committed:
DDP-4127, set up dashboard: changed dashboard title and descriptive text and started working on adding study listing page to dashboard just below activities

charlottedye committed:
DDP-4127, set up dashboard: changed dashboard title and descriptive text and started working on adding study listing page to dashboard just below activities

charlottedye committed:
DDP-4127, set up dashboard: temporarily commented out changes to get to a working state

charlottedye committed:
DDP-4127, set up dashboard: temporarily commented out changes to get to a working state

charlottedye committed:
DDP-4127, set up dashboard: temporarily commented out changes to get to a working state

charlottedye committed:
DDP-4127, set up dashboard: temporarily commented out changes to get to a working state

charlottedye committed:
DDP-4127, set up dashboard: got initial version working for typical case and showing study listing

charlottedye committed:
DDP-4127, set up dashboard: got initial version working for typical case and showing study listing

charlottedye committed:
DDP-4127, set up dashboard: got initial version working for typical case and showing study listing

charlottedye committed:
DDP-4127, set up dashboard: got initial version working for typical case and showing study listing

charlottedye committed:
Prion: fixed errors from rebasing

charlottedye committed:
Prion: fixed errors from rebasing

charlottedye committed:
Prion: fixed errors from rebasing

charlottedye committed:
Prion: fixed errors from rebasing

charlottedye committed:
DDP-4127: finished setting up Prion dashboard: simplified code showing activity listing, fixed various display issues, added created, summary, and actions for study listing, and made activity name column wider than other columns to improve readability

charlottedye committed:
DDP-4127: finished setting up Prion dashboard: simplified code showing activity listing, fixed various display issues, added created, summary, and actions for study listing, and made activity name column wider than other columns to improve readability

charlottedye committed:
DDP-4127: finished setting up Prion dashboard: simplified code showing activity listing, fixed various display issues, added created, summary, and actions for study listing, and made activity name column wider than other columns to improve readability

charlottedye committed:
DDP-4127: finished setting up Prion dashboard: simplified code showing activity listing, fixed various display issues, added created, summary, and actions for study listing, and made activity name column wider than other columns to improve readability

charlottedye committed:
Fixed issue from rebase

charlottedye committed:
Fixed issue from rebase

charlottedye committed:
Fixed issue from rebase

charlottedye committed:
Fixed issue from rebase

charlottedye committed:
Made all frontend text that will need to be translated come from en.json

charlottedye committed:
Made all frontend text that will need to be translated come from en.json

charlottedye committed:
Made all frontend text that will need to be translated come from en.json

charlottedye committed:
Made all frontend text that will need to be translated come from en.json

charlottedye committed:
Prion: Fixed issue where small screen view of dashboard wasn't using the right translation string

charlottedye committed:
Prion: Fixed issue where small screen view of dashboard wasn't using the right translation string

charlottedye committed:
Prion: Fixed issue where small screen view of dashboard wasn't using the right translation string

charlottedye committed:
Prion: Fixed issue where small screen view of dashboard wasn't using the right translation string

charlottedye committed:
Prion: DDP-4151: Fixed small view menu, got join us button to match old Prion, and got sign in/sign out to line up better with other links

charlottedye committed:
Prion: DDP-4151: Fixed small view menu, got join us button to match old Prion, and got sign in/sign out to line up better with other links

charlottedye committed:
Prion: DDP-4151: Fixed small view menu, got join us button to match old Prion, and got sign in/sign out to line up better with other links

charlottedye committed:
Prion: DDP-4151: Fixed small view menu, got join us button to match old Prion, and got sign in/sign out to line up better with other links

charlottedye committed:
Prion: DDP-4145: Footer: added missing links to footer, changed more details to learn more, refactored the translation file, fixed various display issues, got correct buttons to appear depending on user state, cleaned up CSS, fixed issue with FontAwesome icon, etc.

charlottedye committed:
Prion: DDP-4145: Footer: added missing links to footer, changed more details to learn more, refactored the translation file, fixed various display issues, got correct buttons to appear depending on user state, cleaned up CSS, fixed issue with FontAwesome icon, etc.

charlottedye committed:
Prion: DDP-4145: Footer: added missing links to footer, changed more details to learn more, refactored the translation file, fixed various display issues, got correct buttons to appear depending on user state, cleaned up CSS, fixed issue with FontAwesome icon, etc.

charlottedye committed:
Prion: DDP-4145: Footer: added missing links to footer, changed more details to learn more, refactored the translation file, fixed various display issues, got correct buttons to appear depending on user state, cleaned up CSS, fixed issue with FontAwesome icon, etc.

charlottedye committed:
Prion: DDP-4103: Learn more page: changed title, got rid of numbers, changed font and formatting of text, and fixed an issue where the header text wasn't changing color depending on the current page

charlottedye committed:
Prion: DDP-4103: Learn more page: changed title, got rid of numbers, changed font and formatting of text, and fixed an issue where the header text wasn't changing color depending on the current page

charlottedye committed:
Prion: DDP-4103: Learn more page: changed title, got rid of numbers, changed font and formatting of text, and fixed an issue where the header text wasn't changing color depending on the current page

charlottedye committed:
Prion: DDP-4103: Learn more page: changed title, got rid of numbers, changed font and formatting of text, and fixed an issue where the header text wasn't changing color depending on the current page

charlottedye committed:
Prion: DDP-4086: fixed functionality of navigation arrows on homepage

charlottedye committed:
Prion: DDP-4086: fixed functionality of navigation arrows on homepage

charlottedye committed:
Prion: DDP-4371: Moved PDFs to bucket and changed study listing to retrieve each PDF from the appropriate bucket for the environment

charlottedye committed:
Prion: DDP-4371: Moved PDFs to bucket and changed study listing to retrieve each PDF from the appropriate bucket for the environment

charlottedye committed:
Prion: DDP-4371: Moved PDFs to bucket and changed study listing to retrieve each PDF from the appropriate bucket for the environment

charlottedye committed:
Prion: DDP-4371: Moved PDFs to bucket and changed study listing to retrieve each PDF from the appropriate bucket for the environment

charlottedye committed:
Prion: DDP-4371: Fixed issue where substitution of bucket URL was not happening correctly and made all links open in a new tab

charlottedye committed:
Prion: DDP-4371: Fixed issue where substitution of bucket URL was not happening correctly and made all links open in a new tab

charlottedye committed:
Prion: DDP-4349, enable additional medical questionnaires: changed text of study listing summary and button on dashboard

charlottedye committed:
Prion: DDP-4349, enable additional medical questionnaires: changed text of study listing summary and button on dashboard

charlottedye committed:
Prion: DDP-4390, changed error page text

charlottedye committed:
Prion: DDP-4390, changed error page text

charlottedye committed:
Prion: DDP-4390, changed error page text

charlottedye committed:
Prion: DDP-4390, changed error page text

charlottedye committed:
Prion: DDP-4403: Got arrows on home page to line up correctly

charlottedye committed:
Prion: DDP-4403: Got arrows on home page to line up correctly

charlottedye committed:
Prion: DDP-4403: Fixed checkbox alignment in consent form and fixed padding on study-listing page

charlottedye committed:
Prion: DDP-4403: Fixed checkbox alignment in consent form and fixed padding on study-listing page

charlottedye committed:
Prion: DDP-4403: frontend tweaks: created a route that does the equivalent of the join us button to use from Auth0 universal login

charlottedye committed:
Prion: DDP-4403: frontend tweaks: created a route that does the equivalent of the join us button to use from Auth0 universal login

charlottedye committed:
Prion: DDP-4403: frontend tweaks: created a route that does the equivalent of the join us button to use from Auth0 universal login

charlottedye committed:
Prion: DDP-4403: frontend tweaks: created a route that does the equivalent of the join us button to use from Auth0 universal login

charlottedye committed:
Prion: DDP-4403: frontend tweaks: created a route that does the equivalent of the join us button to use from Auth0 universal login

charlottedye committed:
Prion: DDP-4403: frontend tweaks: created a route that does the equivalent of the join us button to use from Auth0 universal login

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: fixed issue caused by rebasing on latest version of develop

charlottedye committed:
Prion: DDP-4482: Added contact address to footer

charlottedye committed:
Prion: DDP-4482: Added contact address to footer

charlottedye committed:
Prion: DDP-4482: Added contact address to footer

charlottedye committed:
Prion: DDP-4482: Added contact address to footer

charlottedye committed:
Prion: DDP-4482: Added contact address to footer

charlottedye committed:
Prion: Fixed config template error

charlottedye committed:
Prion: Fixed config template error

charlottedye committed:
Prion: Fixed config template error

charlottedye committed:
Prion: Fixed config template error

charlottedye committed:
Prion: Fixed config template error

charlottedye committed:
Prion: Fixed config template error

charlottedye committed:
Prion: Addressed some discrepancies between toolkit and prion-toolkit

charlottedye committed:
Prion: Addressed some discrepancies between toolkit and prion-toolkit

charlottedye committed:
Prion: Addressed some discrepancies between toolkit and prion-toolkit

charlottedye committed:
Prion: Addressed some discrepancies between toolkit and prion-toolkit

charlottedye committed:
Prion: Addressed some discrepancies between toolkit and prion-toolkit

charlottedye committed:
Prion: Refactored prion-toolkit to only contain things that should be different from toolkit

charlottedye committed:
Prion: Refactored prion-toolkit to only contain things that should be different from toolkit

charlottedye committed:
Prion: Refactored prion-toolkit to only contain things that should be different from toolkit

charlottedye committed:
Prion: Refactored prion-toolkit to only contain things that should be different from toolkit

charlottedye committed:
Prion: Refactored prion-toolkit to only contain things that should be different from toolkit

charlottedye committed:
Prion: DDP-4434: Support language selection for Prion: worked on getting language selection to work with Prion, including starting to reimplement the study-listing component to translate more cleanly, and did more refactoring.

charlottedye committed:
Prion: DDP-4434: Support language selection for Prion: worked on getting language selection to work with Prion, including starting to reimplement the study-listing component to translate more cleanly, and did more refactoring.

charlottedye committed:
Prion: DDP-4434: Support language selection for Prion: worked on getting language selection to work with Prion, including starting to reimplement the study-listing component to translate more cleanly, and did more refactoring.

charlottedye committed:
Prion: DDP-4434: Support language selection for Prion: worked on getting language selection to work with Prion, including starting to reimplement the study-listing component to translate more cleanly, and did more refactoring.

charlottedye committed:
Prion: DDP-4434: Support language selection for Prion: worked on getting language selection to work with Prion, including starting to reimplement the study-listing component to translate more cleanly, and did more refactoring.

charlottedye committed:
Prion: Re-added SessionWillExpireComponent and reorganized language files

charlottedye committed:
Prion: Re-added SessionWillExpireComponent and reorganized language files

charlottedye committed:
Prion: Re-added SessionWillExpireComponent and reorganized language files

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page:
re-implemented ability to filter full table, fixed appearance of table header, and miscellaneous UI fixes and refactoring

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page:
re-implemented ability to filter full table, fixed appearance of table header, and miscellaneous UI fixes and refactoring

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page:
re-implemented ability to filter full table, fixed appearance of table header, and miscellaneous UI fixes and refactoring

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page:
re-implemented ability to filter full table, fixed appearance of table header, and miscellaneous UI fixes and refactoring

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page:
re-implemented ability to filter full table, fixed appearance of table header, and miscellaneous UI fixes and refactoring

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: added ability to filter by a particular column

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: added ability to filter by a particular column

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: added ability to filter by a particular column

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: a few minor fixes

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: a few minor fixes

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: a few minor fixes

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Got entire table to display correctly and fixed issue where HTML was being interpreted as a string literal instead of as HTML by using innerHTML instead of string interpolation

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Got entire table to display correctly and fixed issue where HTML was being interpreted as a string literal instead of as HTML by using innerHTML instead of string interpolation

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Got entire table to display correctly and fixed issue where HTML was being interpreted as a string literal instead of as HTML by using innerHTML instead of string interpolation

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Fixed issues with html in table cells

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Fixed issues with html in table cells

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Fixed issues with html in table cells

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Fixed issues with html in table cells

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Fixed issues with html in table cells

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Fixed issues with html in table cells

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Worked on re-implementing sorting

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Worked on re-implementing sorting

charlottedye committed:
Prion: DDP-4434, language selection for Prion: Study listing page: Worked on re-implementing sorting

charlottedye committed:
Prion: Fixes for some refactoring issues

charlottedye committed:
Prion: Fixes for some refactoring issues

charlottedye committed:
Prion: Fixes for some refactoring issues

charlottedye committed:
Prion: DDP-4434: Prion language selection: Changed filtering/data source for study listing to use BehaviorSubject instead of Observable to get rid of the need to recreate the data source and re-implemented column filtering

charlottedye committed:
Prion: DDP-4434: Prion language selection: Changed filtering/data source for study listing to use BehaviorSubject instead of Observable to get rid of the need to recreate the data source and re-implemented column filtering

charlottedye committed:
Prion: DDP-4434: Prion language selection: Changed filtering/data source for study listing to use BehaviorSubject instead of Observable to get rid of the need to recreate the data source and re-implemented column filtering

charlottedye committed:
Prion: DDP-4434: Prion language selection: Added component that changes the configured language and redirects to specified destination

charlottedye committed:
Prion: DDP-4434: Prion language selection: Added component that changes the configured language and redirects to specified destination

charlottedye committed:
Prion: DDP-4434: Prion language selection: Added component that changes the configured language and redirects to specified destination

charlottedye committed:
Prion: DDP-4434: Prion language selection: Added component that changes the configured language and redirects to specified destination

charlottedye committed:
Prion: DDP-4434: Prion language selection: Added component that changes the configured language and redirects to specified destination

charlottedye committed:
Prion: DDP-4434: Prion language selection: Miscellaneous refactoring and bug fixes to get the Auth0 pages to work nicely with the language selector/internationalization.

charlottedye committed:
Prion: DDP-4434: Prion language selection: Miscellaneous refactoring and bug fixes to get the Auth0 pages to work nicely with the language selector/internationalization.

charlottedye committed:
Prion: DDP-4434: Prion language selection: Miscellaneous refactoring and bug fixes to get the Auth0 pages to work nicely with the language selector/internationalization.

charlottedye committed:
Prion: DDP-4434: Prion language selection: Miscellaneous refactoring and bug fixes to get the Auth0 pages to work nicely with the language selector/internationalization.

charlottedye committed:
Prion: DDP-4434: Prion language selection: Miscellaneous refactoring and bug fixes to get the Auth0 pages to work nicely with the language selector/internationalization.

charlottedye committed:
Prion: Fixed weird rebase issue that broke the build study script

charlottedye committed:
Prion: Fixed weird rebase issue that broke the build study script

charlottedye committed:
Prion: Fixed weird rebase issue that broke the build study script

charlottedye committed:
Prion: Fixed weird rebase issue that broke the build study script

charlottedye committed:
Prion: Fixed weird rebase issue that broke the build study script

charlottedye committed:
DDP-4773: Added features and bug fixes that support the internationalization of the Auth0 pages.

charlottedye committed:
DDP-4773: Added features and bug fixes that support the internationalization of the Auth0 pages.

charlottedye committed:
DDP-4773: Added features and bug fixes that support the internationalization of the Auth0 pages.

charlottedye committed:
DDP-4773: Added features and bug fixes that support the internationalization of the Auth0 pages.

charlottedye committed:
DDP-4773: Added features and bug fixes that support the internationalization of the Auth0 pages.

charlottedye committed:
DDP-4773: Added features and bug fixes that support the internationalization of the Auth0 pages.

charlottedye committed:
Prion: DDP-4436: Modified padding for language selection dropdown

charlottedye committed:
Prion: DDP-4436: Modified padding for language selection dropdown

charlottedye committed:
Prion: DDP-4436: Modified padding for language selection dropdown

charlottedye committed:
Prion: DDP-4436, persist selected language when participant signs up or logs in: working on getting Pepper components that get text from the backend to update when profile language changes

charlottedye committed:
Prion: DDP-4436, persist selected language when participant signs up or logs in: working on getting Pepper components that get text from the backend to update when profile language changes

charlottedye committed:
Prion: DDP-4436, persist selected language when participant signs up or logs in: working on getting Pepper components that get text from the backend to update when profile language changes

charlottedye committed:
Prion: DDP-4436, persist selected language when participant signs up or logs in: working on getting Pepper components that get text from the backend to update when profile language changes

charlottedye committed:
Prion: DDP-4436, persist selected language when participant signs up or logs in: working on getting Pepper components that get text from the backend to update when profile language changes

charlottedye committed:
Prion: DDP-4436, persist selected language when participant signs up or logs in: working on bug fixes

charlottedye committed:
Prion: DDP-4436, persist selected language when participant signs up or logs in: working on bug fixes

charlottedye committed:
Prion: DDP-4436, persist selected language when participant signs up or logs in: working on bug fixes

charlottedye committed:
Prion: DDP-4436, persist selected language when participant signs up or logs in: working on bug fixes

charlottedye committed:
Prion: DDP-4436, persist selected language when participant signs up or logs in: working on bug fixes

charlottedye committed:
Prion: Misc. minor fixes

charlottedye committed:
Prion: Misc. minor fixes

charlottedye committed:
Prion: Misc. minor fixes

charlottedye committed:
Prion: Misc. minor fixes

charlottedye committed:
Prion: Misc. minor fixes

charlottedye committed:
Prion: DDP-4436, Persist selected language when participant creates account: bug fixes and refactoring

charlottedye committed:
Prion: DDP-4436, Persist selected language when participant creates account: bug fixes and refactoring

charlottedye committed:
Prion: DDP-4436, Persist selected language when participant creates account: bug fixes and refactoring

charlottedye committed:
Prion: DDP-4436, Persist selected language when participant creates account: bug fixes and refactoring

charlottedye committed:
Prion: DDP-4436, Persist selected language when participant creates account: bug fixes and refactoring

charlottedye committed:
Prion: fixed rebase issue

charlottedye committed:
Prion: fixed rebase issue

charlottedye committed:
Prion: fixed rebase issue

MocanaAtBroad committed:
Add public-api.ts file for toolkit-prion

MocanaAtBroad committed:
Add public-api.ts file for toolkit-prion

MocanaAtBroad committed:
Add toolkit-prion to workspace Typescript config paths

MocanaAtBroad committed:
Add toolkit-prion to workspace Typescript config paths

charlottedye committed:
Prion: Fixed rebase issues

charlottedye committed:
Prion: Fixed rebase issues

marat-mamytov committed:
4560 | cut private policy from prion

marat-mamytov committed:
4560 | revert footer change

marat-mamytov committed:
4560 | fix text and classes

marat-mamytov committed:
4560 | return back tcell cookies as necessary functional

charlottedye committed:
Prion: Fixed rebase issues

charlottedye committed:
Prion: Fixed rebase issues

charlottedye committed:
Fixed rebase issues

charlottedye committed:
DDP-4796: Prion: Working on adding privacy policy

charlottedye committed:
DDP-4796: Prion, privacy policy page: Finished initial version of privacy policy text and refactored to use one component for both the modal and the static page

charlottedye commented:
Question on this: 

"For authenticated uses cookies preferences are saved into profile:
New field public cookiesPreferences: CookiesPreferences | null; was added to the UserProfile model.
Need backend support for storing this data."

  I thought we decided not to do this.

charlottedye commented on ddp-workspace/projects/ddp-sdk/src/lib/models/userProfile.ts:
```diff
@@ -7,4 +9,5 @@ export class UserProfile {
     public preferredLanguage: string | null;
     public firstName: string | null;
     public lastName: string | null;
+    public cookiesPreferences: CookiesPreferences | null;
```
I thought we decided not to do the profile support.  If we are doing this, we probably need to fully implement it.

marat-mamytov replied:
Removed saving to user's profile

charlottedye reviewed (changes requested):
I ran the code locally to review the functionality before looking closely at the actual code and found a number of issues: go ahead and fix these and then re-test and check the acceptance criteria to make sure I didn't miss anything.  Also, I thought we decided not to put the cookie preferences in the profile, so we should make sure we're on the same page about the plan for that.

-_gat and _gat_platform sometimes get added before cookies are accepted or rejected
-When the user submits the cookie preferences banner without changing anything (I.e., when they select accept on analytics cookies), the _gid and _ga cookies are added immediately like they should be, but _gat_platform and _gat are added only after doing something like clicking on the header logo.
-Functional cookies and analytical categories are described differently.  Functional cookies are grouped by service and analytical cookies are not.  It would be better to either always group or never group.  At minimum, "Name" should probably mean the same thing for both categories.
-For functional cookies: It seems like we should include the actual name of the cookie here somewhere.  For example, PRION_cookies_consent, tcell_agent_policy_cache, tcell_agent_session_id, etc.  For example, maybe under name we could do something like: "TCell: tcell_agent_policy_cache, tcell_agent_session_id".
-For functional cookies: some groups have expiration listed as "various".  It seems like it would be better to specify the different expirations, either by specifying the expiration for each cookie in the category or just listing the possible expirations.  We could say things in local storage have an expiration of "none"; keep "session" for session storage; and list the actual time before expiration for actual cookies
-For functional cookies: Auth0, TCell, and Pepper platform should probably be capitalized 
-For functional cookies: the descriptions aren't really accurate for some of the categories.  User authorization makes sense for Auth0, although you could also describe it as user authentication or access control.  For Pepper platform, I don't recall what exactly the cookies are (pepper.PRION.irbsession will only be present in test environments so you don't need to include that.  You probably should include PRION_cookies_consent, and that would be for storing cookie preferences).  For TCell, description should say something about security, like website security or just security.
-For analytical cookies: It looks like there's also _gat_platform
-For analytical cookies: the description for _gat should probably indicate that this is for network calls to analytics services.  Maybe something like "throttling to increase the efficiency of analytics collection" (or something shorter if you can think of anything)
-On the welcome page, the join us button at the bottom looks different.  Seems like there was a change to the CSS for Button—primaryYellow.  If something using Button—primaryYellow needs to change, it should do that in a different CSS class

charlottedye committed:
DDP-4796: Prion, privacy policy page: Finished privacy policy and added stub for third party page.

marat-mamytov committed:
Merge branch 'prion_frontend_migration_branch' into DDP-4564_manage-cookie-preferences

marat-mamytov committed:
4564 | move pp button to pp dir

marat-mamytov committed:
4564 | add cookie preference btn to pp page

charlottedye commented:
I thought I had mentioned this but don't see it in my comment, so just in case I left it out: there are at least two missing cookies not described here related to Google Analytics that appear after accepting or rejecting cookies and clicking on the Prion banner at the top.

marat-mamytov committed:
4564 | fix cookies pref modal table

marat-mamytov commented:
> I thought I had mentioned this but don't see it in my comment, so just in case I left it out: there are at least two missing cookies not described here related to Google Analytics that appear after accepting or rejecting cookies and clicking on the Prion banner at the top.

 @charlottedye 
+ I've added a few Pepper cookies stored in local storage: session_key, token, studyLanguage. They are set in SessionMementoService and LanguageService. 
+ Updated the cookies description and table in the cookie preferences modal.
+ Added manage preferences button to the policy page.
+ Fixed Button—primaryYellow styles

_Please check updates if you'll have a chance, how it looks like._

But unfortunately,  I'm not getting _gat and _gat_platform added before cookies are accepted or rejected -> I'll investigate it more. 
And I'm not getting any additional GA cookies (at least two as you mentioned). 

_Could you please remove all cookies and local storage data for prion sites and local storage and check one more time.
If they remain, write their names, I'll investigate it also._

marat-mamytov committed:
4564 | return  gat

charlottedye commented:
I recall at least one case where they showed up was after clicking accept
on the cookie banner and then clicking on the Prion logo in the header.
Try clearing cookies and storage and going to the page, clicking accept,
and then clicking on the Prion logo and see if those show up.  If not, I
can see if it's a configuration issue.  Then make sure things work and you
aren't seeing any cookies you aren't expecting when creating an account,
filling out a survey, logging out, and logging in, etc.  You could also try
searching source control for "_gat", "_gat_platform", and anything related
to analytics.  If you're still not seeing it after that, I can investigate
and that would probably mean either it's a configuration issue or it was
inadvertently fixed somehow.

On Wed, Aug 12, 2020 at 12:18 PM Marat <notifications@github.com> wrote:

> I thought I had mentioned this but don't see it in my comment, so just in
> case I left it out: there are at least two missing cookies not described
> here related to Google Analytics that appear after accepting or rejecting
> cookies and clicking on the Prion banner at the top.
>
> @charlottedye <https://github.com/charlottedye>
> I've added a few Pepper cookies stored in local storage: session_key,
> token, studyLanguage. They are set in SessionMementoService and
> LanguageService.
>
>    - Updated the cookies description and table in the cookie preferences
>    modal.
>    - Added manage preferences button to the policy page.
>    - Fixed Button—primaryYellow styles
>    Please check if you'll have a chance.
>
> But unfortunately, I'm not getting _gat and _gat_platform added before
> cookies are accepted or rejected -> I'll investigate it more. I don't even
> have _gat cookie at all for some reason.
> And I'm not getting any additional GA cookies (at least two as you
> mentioned). Could you please write their names, I'll investigate it also.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/broadinstitute/ddp-angular/pull/215#issuecomment-672973468>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJPNM2J2VB2ZZ7J3DV2LQETSAK6FVANCNFSM4PVMACAA>
> .
>

marat-mamytov committed:
4564 | fix _gat and _gat_platform issues

marat-mamytov committed:
4564 | add pepper session cookies from Auth0AdapterService

marat-mamytov committed:
4564 | add auth0 cookies, they are set only during signin/up

marat-mamytov committed:
4564 | add translations

marat-mamytov commented:
@charlottedye 
> Also, I thought we decided not to put the cookie preferences in the profile, so we should make sure we're on the same page about the plan for that.

Misunderstood, fixed, no saving into profile.

> -_gat and _gat_platform sometimes get added before cookies are accepted or rejected
> -When the user submits the cookie preferences banner without changing anything (I.e., when they select accept on analytics cookies), the _gid and _ga cookies are added immediately like they should be, but _gat_platform and _gat are added only after doing something like clicking on the header logo.

Fixed my local issue with _gat cookie. Thank you! =>  Now both _gat and _gat_platform are set **with** _ga and _gid on acceptance. And no _gat and _gat_platform cookies are added before consent.
![image](https://user-images.githubusercontent.com/54633385/90141116-825d2300-dd9c-11ea-9383-609a911b8218.png)

> -On the welcome page, the join us button at the bottom looks different. Seems like there was a change to the CSS for Button—primaryYellow. If something using Button—primaryYellow needs to change, it should do that in a different CSS class

Fixed.

Also fixed cookies list and descriptions.

charlottedye reviewed (changes requested):
I tried to rebase this on the latest version of the Prion branch multiple times but it does not seem to be working.  It might be better if you do that or even start fresh with a new branch based on prion_frontend_migration_branch and put all your changes in one commit since this is all one ticket anyway.

Additionally, I ran the non-rebased version of this locally and there are some basic UI things in the cookie preferences window that appear to have been messed up at some point.  I'm assuming the extra space under the button and link at the bottom of the modal window for the your privacy tab was unintentional because it looks a little strange to me.  At at least one screen size (I noticed this at 1121px x 721px), the bottom section with the link and button starts at a different place on the modal on each tab.  Looks like I might have missed this on the initial review.  Of the three tab versions, I think the analytical cookies tab's version looks the best, personally, but pick something that looks reasonable and use it for all three.  

Nice work finding the additional cookies.
